redis_json/src/c_api.rs

```diff
@@ -267,6 +272,54 @@ pub fn json_api_is_json<M: Manager>(m: M, key: *mut rawmod::RedisModuleKey) -> c
     m.is_json(key).map_or(0, |res| res as c_int)
 }
 
+pub fn json_api_get_key_value<'a, M: Manager>(_: M, val: *const c_void) -> *const c_void
+where
+    M::V: 'a,
+{
+    let json = unsafe { &*(val.cast::<M::V>()) };
+    match json.get_type() {
+        SelectValueType::Object => {
+            let vec = json.items().unwrap().collect::<Vec<(&'a str, &'a M::V)>>();
+            Box::into_raw(Box::new(KeyValuesIterator {
+                results: vec,
+                pos: 0,
+            }))
+            .cast::<c_void>()
+        }
+        _ => null(),
+    }
+}
+
+pub fn json_api_next_key_value<'a, M: Manager>(
+    _: M,
+    iter: *mut c_void,
+    ctx: *mut rawmod::RedisModuleCtx,
+    str: *mut *mut rawmod::RedisModuleString,
+) -> *const c_void
+where
+    M::V: 'a,
+{
+    let iter = unsafe { &mut *(iter.cast::<KeyValuesIterator<M::V>>()) };
+    if iter.pos < iter.results.len() {
```

Comment:
I would prefer:
```rust
if iter.pos >= iter.results.len() {
    return null();
}
```

Reply:
Even if this if will usually be not-taken?

Reply:
Why is it different? In both cases you are doing an if statement.

Reply:
But most times the if will not be taken? Shouldn't we prefer when we have some info, to put the more-taken branch in the `if`, and the less-taken branch in the `else`?

Reply:
Why do we prefer this? Are you afraid of performance because of branch prediction? Maybe we can use [unlikely](https://docs.rs/likely_stable/latest/likely_stable/fn.unlikely.html)?

Reply:
Avoiding the else